Wikipedia/Code/ArticleViewController+Editing.swift

```diff
@@ -76,24 +102,142 @@ extension ArticleViewController {
     }
     
     private func presentEditor(editorViewController: UIViewController) {
-        
-        let navigationController = WMFComponentNavigationController(rootViewController: editorViewController, modalPresentationStyle: .overFullScreen)
-        
-        let needsIntro = !UserDefaults.standard.didShowEditingOnboarding
-        if needsIntro {
-            let editingWelcomeViewController = EditingWelcomeViewController(theme: theme) {
+        let presentEditorAction = {
```

Comment:
Can you update this to:

```
let presentEditorAction = { [weak self] in
      guard let self else { return }
      ...
}
```

I noticed the new temp accounts views weren't deallocating and this might help.